src/vs/workbench/contrib/tasks/browser/taskTerminalStatus.ts

```diff
@@ -50,29 +52,27 @@ export class TaskTerminalStatus extends Disposable {
 	addTerminal(task: Task, terminal: ITerminalInstance, problemMatcher: AbstractProblemCollector) {
 		const status: ITerminalStatus = { id: TASK_TERMINAL_STATUS_ID, severity: Severity.Info };
 		terminal.statusList.add(status);
-		this.terminalMap.set(task, { terminal, status, problemMatcher });
+		this.terminalMap.set(task._id, { terminal, task, status, problemMatcher });
 	}
 
 	private terminalFromEvent(event: TaskEvent): TerminalData | undefined {
-		if (!event.__task || !this.terminalMap.get(event.__task)) {
+		if (!event.__task || !this.terminalMap.get(event.__task._id)) {
```

Comment:
`this.terminalMap.get(event.__task._id)` should only be `TerminalData | undefined` so you could leave this out of the if as it's returned immediately below.